.gitignore

```diff
@@ -66,3 +66,7 @@ htmlcov
 # vi / vim swp files
 *.swp
 .DS_STORE
+
+# setuptools_scm files
+# this should be generated automatically when installed
```

Comment:
Does this include `python setup.py develop` ?

Reply:
Anything that runs `setup.py` *should* generate the necessary version files, however `python setup.py X` is technically deprecated. The preferred way of installing things is `pip install .` (full install) or `pip install -e .` (development mode).

However-however, `pip install -e .` won't work if we add a `pyproject.toml` without adding a `--no-use-pep517`. This is because I guess development mode is also deprecated?

Reply:
https://twitter.com/pganssle/status/1152694289920679936

Reply:
ok, but does this version.py get generated with `pip install -e . ` ?